Examples/WindowsFormsApplication/Nfield.cs

```diff
@@ -0,0 +1,138 @@
+﻿using Microsoft.Identity.Client;
+using Microsoft.Win32;
+using Nfield.Infrastructure;
+using Nfield.Models;
+using Nfield.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace WindowsFormsApplication
+{
+    public class Nfield
+    {
+        private IPublicClientApplication ClientApp { get; init; }
+        private string[] Scopes { get; } = new[] { CreateScope("user_impersonation"), CreateScope("Survey.Read") };
+        public string DomainName { get; }
+
+        private INfieldConnectionV2 NfieldConnection { get; set; }
+
+        public Nfield(string domainName)
+        {
+            ClientApp = CreateClientApp();
+            DomainName = domainName;
+
+            NfieldConnection = NfieldConnectionFactory.Create(ApplicationConfiguration.Current.NfieldApiUri);
+            NfieldConnection.RegisterTokenProvider(domainName, ProvideTokenAsync);
+        }
+
+        private IPublicClientApplication CreateClientApp()
+        {
+            var client = PublicClientApplicationBuilder.Create(ApplicationConfiguration.Current.ClientId)
+                .WithRedirectUri("http://localhost")
```

Comment:
This is a native client right? do they need redirecturi's to localhost? I thought that works differently (with some weird strings and stuff)

Reply:
What is also funny here is that you enabled the "native client platform" on the app registration yet in this appbuilder nothing points to that it is a native client (that cannot  keep secrets). Not super familiar with this new apis but it made me scratch my head a bit :-D

Reply:
The readme explains why this URI is used, including links to ms docs.